### ISSUE TITLE
Introduce yourself!

### ISSUE THREAD

**Issue by welschp** [WhiteHouse/ndoch-hackathon#1]
Welcome to the repo for the National Day of Civic Hacking at the White House! We're pulling together a great team for this event, including folks who'll join us here at the White House on June 1 and others who'll be taking part remotely all through the month of May. 

I thought I'd open a thread so everyone can introduce themselves and describe the kind of projects they're interested in working on. I'll go first.

My name's Pete Welsch, I'm Deputy Director of Online Platform in the White House Office of Digital Strategy. I work with the teams of developers behind We the People and WhiteHouse.gov. If you have any questions or concerns about the hackathon or the We the People API, please don't hesitate to reach out.

Who's next?

**Comment by waldoj** [comment]
Howdy! I'm Waldo Jaquith, the developer of [The State Decoded](/statedecoded/statedecoded/), an open-source application to put legal codes online, making them understandable, and put an API on top of them. I've been a website developer for 20 years, dabbling in open government data for 18 years, and making a full-time job of open government data for the past few years now.

**Comment by jmandzik** [comment]
Hi!  I'm [Justin Mandzik](http://mandzik.org), a senior software engineer at Time Warner Cable.  Most of my days are spend writing realtime web applications for network performance & fault monitoring.  Personally and professionally I love writing data visualizations that make sense of big data sets and geospatial information.  Any kind of realtime mashup is fair game and fun to do.  If anyone else is going to be in attendance on June 1st and wants to collaborate, please feel free to reach out to me at justin.mandzik@gmail.com.

**Comment by ivanstegic** [comment]
Hello, I'm [Ivan Stegic](http://twitter.com/TEN7ivan) and I run a small web development studio in Minneapolis called TEN7. We're focused on developing Drupal sites, but we really just like solving problems with whatever tools we have. We end up using Google APIs quite a bit, especially Maps V3 and Fusion Tables. I'm interested in finding out what the timelapsed (and real time) lifespan of a petition looks like geo-spatially; what all petitions look like on a map; and others. I'm curious to know if a petition is a local thing that goes regional, then national; what a petition looks like when it goes viral or if they're totally random.

**Comment by ivanstegic** [comment]
I just realized that the [API Gallery](https://petitions.whitehouse.gov/how-why/api-gallery) has two map implementations, [one of which](https://github.com/dthompson/WhereThePeople) is close to what I was thinking about... let the hacks begin!

**Comment by jessebeach** [comment]
I'm [Jesse Beach](https://twitter.com/jessebeach) from Boston. I work for [Acquia](http://acquia.com), primarily building front end UIs for Drupal 8. JavaScript is my strongest language, followed by CSS. I focus on accessibility as well. I'm decently skilled with PHP, but I prefer to stick with front end code. My ideal day hacking is teaming up with a stellar PHP-dev/Rubyist/Java/whatever and working in parallel on data handling and UI.

ivanstegic, I had the same idea to set up some kind of map vs. time display, but it seems those types of projects are already underway. I challenge us to come up with some clever twist on the idea or perhaps something new entirely!

jmandzik, I'll be in DC on June 1st. It seems like our skills would match up well.

**Comment by jessebeach** [comment]
Hi waldoj! Sorry, I would have said hello in my previous comment, but your message was scrolled off my page and I didn't notice until just now. Looking forward to building something with you perhaps!

**Comment by heyitsgarrett** [comment]
Hi there, [Garrett Miller](http://twitter.com/heyitsgarrett) – I'm a developer and designer at [MapBox](http://www.mapbox.com), which I joined recently after several years with [Opower](http://www.opower.com). I'm friends with a few folks who participated in the first hackathon and am eager to join the fray. 

It seems like we're already well into interesting ways of mapping this data and I'm happy to lend my experience and resources to the matter. @ivanstegic and I might have some things to think about!

**Comment by stevegrunwell** [comment]
Hi, I'm [Steve Grunwell](http://stevegrunwell.com/), a developer with [Buckeye Interactive](http://www.buckeyeinteractive.com/) in New Albany, Ohio.

I do a lot of work with and am a big fan of WordPress so I'm focusing my efforts on a plugin that allows authors to easily embed petition information into posts. When the write API is released sometime in the vague future I plan to update the plugin to accept signatures right within the post - how cool would it be to read someone's blog post on an issue, follow the discussion in the comments, then sign the petition all from a single screen?

I'll be in D.C. on the 1st, I look forward to meeting all of you!

**Comment by jmandzik** [comment]
Seems we all love maps!

I'm looking into the feasibility of showing some geospatial & accompanying visualizations showing signature acceleration.  How fast (and where) are petitions going viral, so-to-speak.  Potentially useful as a way to identify hot topics that may not have the total signature count that would traditionally indicate "importance".  Then, poll the API as often as they'll let me to show realtime activity of the most active petitions.  Welcoming anyone who wants to look into it with me.  And since @heyitsgarrett is coming, I'll have to use mapbox instead of Google Maps :)

**Comment by ccohn** [comment]
Hi, I'm Corinna Cohn, a web developer for Science Magazine. I've created a few Protovis visualizations in the past (map, population pyramid). jmandzik, your idea is exactly what I proposed in the initial registration form, so if you're looking for a team up on that I am definitely interested.

**Comment by danielsmc** [comment]
Hi, I'm Daniel McLaughlin, a creative technologist at the Boston Globe. It looks like a lot of us are excited about the geodata— one way of building off that which could be interesting would be using it to calculate similarity between petitions. Do petitions that are signed by citizens with similar geographical distributions have meaningful relationships that can't be discovered otherwise?

I'll be at the White House hackathon, and I'm looking forward to building something cool with you all!

**Comment by lheyman** [comment]
Hi All, I'm Leigh Heyman, Director of New Media Technologies here at the White House. I oversee all the web operations and software development programs, and coordinate with Pete and others within the Executive Office of the President to help set the technical agenda and project priorities.

While I don't anticipate playing much of a role in the hackathon itself, I look forward to meeting everyone, and catching up with some familiar faces from our last hackathon, and generally excited to see all these great ideas come to life over the coming weeks.

Echoing what Pete said, please don't hesitate to reach out to us with any questions or concerns.

**Comment by jm** [comment]
Hello!  I'm Jeremy McAnally, and I work at GitHub.  We produce this fine piece of software goodness that you're currently using to read this.  I attended the last hackathon and produced [an iPhone app using RubyMotion](http://github.com/jm/WeThePeople) using [the RubyGem I wrote previously](http://github.com/jm/we_the_people).

I'm fairly flexible and polyglot, so if there's an idea in need of hands from a backend or mobile developer, count me in.  I'd love to try to nail down the project I'll be working on so I can hammer out any additions to my gem or new libraries that'll reduce friction on the day of the hackathon.

I love all the data relation-y type ideas here; I think they're fantastic.  I'd love to see some topical relations going on as well as geographic.  But what I'd love to see also is a _really_ solid iPad app that would eventually evolve into a "mobile activist" app that would be used to collect signatures.  Currently, since there's no write API, we'd just be pulling the petitions and such, but having the infrastructure in place (e.g., a CocoaPod-powered lib that we can easily update and distribute to other mobile developers) would be huge.

Of course, if there's little interest in that, I'm up for whatever.  :smile:  Just ping me via e-mail (see my GitHub profile) or what not if you have any specific requests or questions!

**Comment by douglasback** [comment]
Hi everyone! I'm Douglas Back, and I work at [Blue State Digital](http://www.bluestatedigital.com) as a front-end developer. Like Jeremy, I was also at the last hackathon, and built [Widget the People](https://github.com/douglasback/widget-the-people), a Node.js app that generates embeddable widgets that promote petitions on We the People.

My mapping skills are… mediocre, so I think I'll stay focused on building tools that allow petition creators and supporters to promote petitions across the web. I'm leaning heavily towards building a dashboard that helps petition creators keep track of their petition's progress on We the People; monitors mentions of the petition on the web and across social networks; tracks how many times it's been shared or mentioned; and gives users insights into where their petition may be doing well (or not so well). Of course, plans may change between now and June 1, but that idea won't get out of my head. Whatever I wind up working on, I'll be collaborating with @yahelc to build it, who I'll let introduce himself. And if this sounds like something you might be interested in pitching in on, definitely let us know.

Looking forward to meeting all of you on June 1 and seeing what everyone builds!

**Comment by ivanstegic** [comment]
Looking forward to meeting you all in person too. I created [a public list](https://twitter.com/TEN7ivan/wh-civic-day-hackers/members) of twitter handles from this thread I could figure out, so please publish your handle if you're not on there and you'd like to be: @danielsmc @ccohn @lheyman yours eluded me.

@heyitsgarrett would love to try my hand at implementing Mapbox -- are there any facilities for a time lapsed plot of points? There's really no easy way to do it with GMaps V3 and @jmandzik would love to see the acceleration of the signature process too. Yes on collaborating on that then?

**Comment by heyitsgarrett** [comment]
@ivanstegic - Absolutely, re: time-lapsed points. We've been working pretty hard on our newest release (currently v1.0.2) of [mapbox.js](http://mapbox.com/mapbox.js/api) which is based on [Leaflet](http://leafletjs.com). Most things are possible through some clever usage of JSON and JavaScript.

**Comment by nickcatal** [comment]
Hi all! My name is Nick Catalano and I work as a developer/tech guy at the [New Organizing Institute](http://neworganizing.com/), a nonprofit in DC that holds trainings New Media and Data best practices as well as does various work in the Election Administration space.

This is my first hackathon, but a few weeks ago I independently built a [petition response viewer](http://petitions.nickc.at/) hosted on Heroku (and [open source](https://github.com/nickcatal/petitionviewer)) that scrapes the 'We The People' responses nightly, processes the data using the python library beautifulsoup and displays them in a more user-friendly way.

I'm excited to see what with others are doing and if they need help but I'm personally looking at ways of building a very basic python SDK of sorts on the API that means you can write very simple requests to the We The People site with a simple python package. I've already built a [salesforce wrapper](https://github.com/neworganizing/simple-salesforce) so I may adapt those concepts to the We The People site. Imagine a python programer simply having to type `response = wh.Petitions.get('abcd123')` to create a dictionary of the JSON response for petition `abcd123`!

Look forward to meeting everyone.

**Comment by ogglodyte** [comment]
Geospatial views of petition signature patterns are interesting, but I'd think the greater insight would come from visualizing a mashup with Census/lifestyle data. For those that self-report a zip code we get good granularity. Those self-reporting only a state, less so.

There's a sample data set we can play with. I'm dabbling with using Mallet to topic model petition titles and bodies. Could be linked to a geospatial view, but also interesting to see if the topic/keywords that emerge from titles differ from petition bodies (which, at first blush, appears to be the case). There's other fun to be had looking at topics, title/body agreement, and signature rates (what's the benefit of clear/consistent writing). Last bit on this is whether petitions can be topic-grouped - do petitions suffer dilution of signature rates? What are the characteristics of a petition that gets signatures vs one that doesn't if we look only at petitions on a single subject or subject group. How might someone visualize that so that future petition authors can make use of it?

Last idea for the day is how to visually mashup current events with petition submissions and signature rates. Gun control petitions before and after Sandy Hook for instance.

**Comment by ivanstegic** [comment]
@ogglodyte all good points. I'd be especially interested in the before/after affects in certain regions. I do think that a start for this is a map that is able to plot the petitions as a whole, then allow for user filtering by category, then the addition of time laps information per petition. Adding layers with other data could be added as a next step. Will you be there in person on 6/1?

**Comment by ogglodyte** [comment]
@ivanstegic - yes, I'll be there 6/1. Agreed that getting the basic framework put together and then adding layers as time/skills/availability permit is a good approach. Also thinking through some non-geo visualizations. It's a shame that signatures are only identified by a pair of initials - some network/relational analysis would be very cool to understand serial-signers (they sign everything) and to look for astro-turfing, but I fear the available data simply won't support it.

**Comment by ivanstegic** [comment]
@ogglodyte what's your twitter handle? would like to add you to [the list](https://twitter.com/TEN7ivan/wh-civic-day-hackers/members)

**Comment by ivanstegic** [comment]
@ogglodyte i wonder if we could aggregate the data by making some assumptions like same initials in the same zipcode with serial signatures closely spaced together in time? Might be crude, but might yield something?

**Comment by mjangda** [comment]
Howdy! I'm Mohammad Jangda (usually go by Mo) and I work as a Code Wrangler at <a href="http://automattic.com">Automattic</a>. I work on the <a href="http://vip.wordpress.com">WordPress.com VIP team</a> where I help companies like TIME and CNN do cool things with WordPress. I also contribute to the  WordPress open source project and have frequented a hackathon or two in the past.

@stevegrunwell we should chat :)

**Comment by cndreisbach** [comment]
Hi! I'm Clinton Dreisbach ([twitter](https://twitter.com/cndreisbach)) ([github](https://github.com/cndreisbach)), and I am a Technology Fellow at the [Consumer Financial Protection Bureau](http://consumerfinance.gov). I mainly work on backend projects, including [a data platform to serve up large datasets](https://github.com/cfpb/qu) and [tools to work with the Home Mortgage Disclosure Act data](https://github.com/cfpb/hmda-tools).

I would be super happy to work on a project someone else is excited about that I can provide data analysis or hardcore backend support on.

**Comment by cndreisbach** [comment]
@douglasback Your idea looks ambitious and exciting. I would love to help out any way I can.

**Comment by jmandzik** [comment]
I love the talent pool in this thread.  At first glance, there appears to be a couple topical interests resonating, mainly quantitative spatial analytics (map/dashboard stuff), user engagement (mobile app, blog plugins), qualitative analytics (topical trends).  Since I'm talking to folks via email, twitter, and here, I took the liberty in creating an issue in this repo with my area interests to publicly track ideas with whomever may interested.

Short version, I'm looking into census/WTP mashups as an analytical tool.  Tracking progress [here](https://github.com/WhiteHouse/ndoch-hackathon/issues/2).

**Comment by jessebeach** [comment]
@douglasback and @yahelc, I started an [issue](https://github.com/WhiteHouse/ndoch-hackathon/issues/3) for the dashboard idea. I'm interested in working on this with you.

**Comment by eshtadc** [comment]
Hello everyone!  I'm Lisa Backer.  I work here in DC at a small consulting firm (Fig Leaf Software) and I'm a contractor with the Office of Digital Design and Innovation at the Broadcasting Board of Governors (think Voice of America).  I'm really excited to be involved in this and read everyone's ideas.  My interests in the We the People data lies in  bringing the relevant petitions based on interests metadata to people who are passionate about those interests.  I'd like to tie it to new aggregation data.  I'm also interested in taking petition by issue information and analyzing it along with budget data of that particular issue side by side.

**Comment by ogglodyte** [comment]
Next wacky idea... mash up news stories and petitions. Do petitions follow reporting? Does reporting follow petitions? Are they concurrent? How likely is a petition related to current events to hit a signature threshold? How long/popular does the reporting have to be in order to drive signatures?

It's still along the lines of topic modeling, which is no small nut to crack in itself, but there might be some reasonable ways to narrow things down, do some keyword kludges etc. For news feeds, there's screen scraping blogs and the like (is Huffington searchable?). Another idea is here (http://www.npr.org/api/inputReference.php) -- NPR's story API. Anyone know of other, similar, news APIs?

And now that I think about it, there's also the mash-up with tweets - how do twitter activity and petitions relate, etc.

**Comment by ivanstegic** [comment]
Fantastic pie in the sky idea! NY Times has an incredible API implementation, perhaps something like this to start? http://developer.nytimes.com/docs/read/article_search_api_v2

**Comment by ogglodyte** [comment]
There are some really sweet D3-based timeline visualizations that could be co-opted for this one

**Comment by ogglodyte** [comment]
Last one for the day - explore how the White House responds to petitions that receive the requisite signatures. The API provides links to response pages. The responses themselves often (always?) have links to other pages - policy statements, legislative summaries, whatever. It might be fun to make a node-link diagram showing petition responses and the sites/pages they link to. Do certain authors always link to the same stuff? Do all responses link to common pages? Mouse over a node to see the page title...

**Comment by jessebeach** [comment]
ogglodyte, oh, I like that a lot. Great idea!

**Comment by heyitsgarrett** [comment]
@ivanstegic @ogglodyte - I actually played with the NYTimes API for somewhat similar reasons. It's a rough prototype, but I've hosted it here:

[World News Pulse](http://heyitsgarrett.com/share/pulse)

Would love to find a way to combine the two APIs. Great idea.

**Comment by carldanley** [comment]
I'm Carl Danley and I'm a Senior Web Engineer for [10up](http://www.10up.com). We are a premium web development agency focusing specifically on WordPress. We do a lot of work with [WordPress VIP](http://vip.wordpress.com). I'm the [140th](https://github.com/jquery/jquery/blob/master/AUTHORS.txt#L140) person to contribute to jQuery core and I contribute code to various WordPress plugins and have multiple pending tickets for WordPress core. I primarily focus on architecting efficient JavaScript applications with a heavy emphasis on performance and well-written, modularized code. I've had the opportunity to work on various large brand names such as [TechCrunch](http://techcrunch.com), [Global News](http://globalnews.ca) and [Godiva](http://godiva.com) to name a few.

I will be coordinating with @mjangda and @christophercochran to deliver a WordPress plugin that uses the We The People API to embed petitions into WordPress sites. The plugin will offer a variety of unique views among other cool features we will unveil at the hackathon.

I'm looking forward to meeting you all and having the opportunity to hack on some awesome stuff in the White House! Also, fun facts: I love Mountain Dew and long walks on the beach =P

**Comment by stevegrunwell** [comment]
@carldanley It sounds like you, @mjangda, @christophercochran, and I had the same thought. I'm also working on a WordPress plugin that embeds petitions via shortcodes and widgets. My repo's started over at https://github.com/buckii/we-the-people-wordpress/. Want to pool resources?

**Comment by nickcatal** [comment]
I was personally thinking it'd be great to have a system that holds the White House accountable by counting the number of days between a signature passing their mark and getting a response.

And then once a response is posted it will display the response, next to the days between the signature threshold being crossed and the response.

Would probably be relatively straightforward to do. Just pull from the petitions endpoint nightly and then the frontend just displays the count.

Anyone interested?

**Comment by ivanstegic** [comment]
@carldanley that Wordpress plugin sounds appealing! We're a Drupal shop, perhaps we could co-ordinate and I could build a Drupal module with similar features? Soooo many potential things to work on!

**Comment by eshtadc** [comment]
@ivanstegic @ogglodyte @heyitsgarrett I'd love to work with you on something combining the news trends and petition data.  Would you like to open an issue and get started based on what APIs are available and looking at the prototype by @heyitsgarrett ?

**Comment by ogglodyte** [comment]
@lkhaas: in. I'm dabbling with a small toy in D3 to visualize signatures and petitions over time. Limited spare time, but if I get something working I'll share.

In other news, here's an interesting read: http://www.theatlantic.com/china/archive/2013/05/new-thing-chinese-web-users-love-petitioning-the-white-house/275648/

Anyone up to the challenge of trying to identify petitions from foreign sources?

**Comment by ivanstegic** [comment]
I think we'd be able to weed out foreign petitions if the API contained IP address info. I wonder if that could be captured and associated with the signature if it isn't already. At least we could then filter out non Tor, non proxy petitions that originate outside the US
—

**Comment by fuhteems** [comment]
Hi, my name is Fatimah Kabba and I'm a Visual Designer and Strategist from Boston, MA. I don't have the coding prowess that many of you do, but I have a long standing love for political issues and social activism. 

I'm excited to see how we can help people bolster their petitions with articles, infographics, and data visualization whether they are generated programmatically or pulled in from the web via tags and keywords.

Individuals creating petitions definitely need a repository of information to reign in, inform and retain followers. The current API is an amazing tool, that can be expanded upon by allowing users to bolster their petitions with references. I'd love to help build an experience that has a visually engaging, informative interface that comprehensively breaks down key issues for the user in a way that can then be shared with their audience.

I'm really excited to a part of this, and am ready to help out in any way I can. I can't wait to meet everyone!

**Comment by bryanbraun** [comment]
Hello. My name is Bryan Braun and I work on the technology team that helped develop the API. I will be attending the Hackathon and I'm planning on joining one of your teams and assisting in whatever way I can.

My experience is in the front-end, so I'll be most useful if working with HTML, CSS, and Javascript, (though I know a bit of PHP as well). Because I was not directly involved in developing the API, I cannot provide deep insight regarding the API itself -- most of what I know is already in the [online documentation](https://petitions.whitehouse.gov/developers). Regardless, I'm looking forward to working along-side you all and making something awesome.

**Comment by carldanley** [comment]
@bryanbraun Would love to work with you at some point during the day so we can chat about some API-based improvements.

**Comment by roberte3** [comment]
Hello, I'm Rob Eickmann I was lucky enough to get to participate in the first hackathon, and wrote an iPhone app the runs on the API. (You can download it from apple appstore, at get the source code at http://wethepeopleapp.us) 

I will be participating from Seattle at the Hack for Change event on the 1st. (http://www.meetup.com/Code-for-Seattle/events/113040382/)

My goal for the hackathon will be to enable push notifications in the iPhone app. (Probably running on an Azure node.js backend).

**Comment by ivanstegic** [comment]
@bryanbraun @carldanley Would love to participate in that as well... curious to see how Services is cofigured for the API. Assuming it's Services :)

**Comment by roacobb** [comment]
My name is Rodney(@facetherathe) I'm a current mobile developer and data analyst working Washington DC. Very much looking forward to the day and being at the White House. I'm going to be using an agency API and do some analysis and data viz of the recent USGS data published, http://waterdata.usgs.gov/nwis/qw, on water quality for the nation.

**Comment by bryanbraun** [comment]
@carldanley It would be great to get your feedback and requests for the API. A good place to submit bug reports and feature requests for the API specifically is the [petitions project on Github](https://github.com/WhiteHouse/petitions), since the API buildout is contained in that project. As such, @ivanstegic, if you or anyone else is curious about the API implementation, you can always look at [the code for the petitions project](https://github.com/WhiteHouse/petitions) since it is publicly available. Of course, there will be plenty of opportunities to discuss the API at the Hackathon as well (again, others on my team are more knowledgable about the details of it than I am).

**Comment by bryanbraun** [comment]
On the topic of hackathon project brainstorming, one thought I had is that if a team is working on some sort of embeddable tool for people to put on their blogs, etc, you could also find a way to integrate that work with the public petitions project via pull request. For example, it might be neat if under each petition, there was an "embed this" button that output the code for your embeddable.

Could be a neat integration for the work being done by @mjangda, @carldanley, @stevegrunwell, or other teams.

**Comment by heyitsgarrett** [comment]
For those interested and in town the night prior to the hackathon, MapBox would love to host you for dinner, drinks, and general idea-swapping. This is unofficial and not affiliated with the White House. See #8 for more details!

**Comment by dlapiduz** [comment]
Hi there, I am Diego Lapiduz (@dlapiduz) from Boca Raton, FL. I work with @cndreisbach at the CFPB and I am really looking forward to meeting you all.

Regarding to what I am planning to build, I am thinking of building an web app where you can see petitions in relation to what your congressman is working on. I know this is not apples to apples but it would be interesting to allow people to communicate to congress what they are supporting or petitioning in We The People.

**Comment by colmdoyle** [comment]
Hey All, 

I'm Colm Doyle, a Software Engineer at [Facebook](https://www.facebook.com) who works on our efforts around Non-Profits and Government. I'll be at the hackathon in person to help out if you plan to integrate any element of [Facebook Platform](https://developers.facebook.com) into your projects.

If you're hacking remotely, I'll also be able to take questions on the day too, probably through an issue here on the repo.

Looking forward to seeing what everyone comes up with!

Colm
http://fb.me/colmdoyle
http://twitter.com/colmisainmdom

**Comment by yoni** [comment]
Hello, I am [Yoni Ben-Meshulam](http://heyitsyoni.com), a Data Scientist with Opower. I participated in the first hackathon and will be attending on June 1, as well. I'll also take a little credit for roping @heyitsgarrett into joining us this time. :)

For the last hackathon, I wrote the [r_we_the_people](https://github.com/yoni/r_we_the_people) R package. I'm not sure what I'll be working on this time, but it will likely fall into one of these buckets:
1. Extract interesting features (e.g. entities, locations, sentiment, writing style) from the petition text and make them available via REST endpoints. This might be done at the individual petition level or across issues. I'll probably use one or more of these APIs to do the heavy lifting: [AlchemyAPI](http://www.alchemyapi.com/), [Open Calais](http://www.opencalais.com/), [uClassify](http://www.uclassify.com/), [Semantria](https://semantria.com/)
2. Create a predictive model for which petitions will get signatures (my stretch goal from last time)
3. Serve up visual assets of individual petitions and issues for other apps to use. Similar to what I did with the [r_we_the_people](https://github.com/yoni/r_we_the_people) package examples

I am comfortable across the stack, including Java, Ruby, Python, R, and JavaScript.

I love all of the great ideas flowing here!! Looking forward to meeting y'all.

Best,
Yoni
[@spaceborne](https://twitter.com/spaceborne)

**Comment by mnloff** [comment]
Hi folks!

I'm Matt Loff [@mnloff](http://twitter.com/mnloff), and I'm the co-owner of [Visionist, Inc.](http://www.visionistinc.com) - a small Maryland-based Dept. of Defense consulting firm.  We specialize in data analytics, signal processing, and visualization.

I threw together "[We the Politics](http://wtp.visionistinc.com/)" for the lack hackathon, and had a blast doing so.  

Yoni and I are talking over ways of collaborating this time around, and I plan on reusing a lot of my code from the first go-round.  As soon as I incorporate a few bugfixes into the API client, I'll be pushing the whole source repo to Github as well.

For June 1, my goal is to build more of a koisk-type web application -- one that relies on a semi-intelligent backend service to pull new petitions & signatures from the white house on a recurring basis, then buffer + stream them out to clients, making the experience feel more like a live Twitter stream or chatroom, as opposed to a collection of queryable pages.

Looking forward to meeting you all soon!

Cheers,
Matt

**Comment by typesend** [comment]
Greetings, my fellow hackers! (And a special shout out to @ivanstegic, as we are both traveling from Minnesota.)

Technologies I like using lately include: Ruby on Rails, Go, Riak, and Postgres. Currently my day job includes developing an internal tool for Symantec, but they were kind enough to let me work remotely for several days surrounding this event.  I plan to be exploring the D.C. area thru Wednesday, in a mediocre attempt to make the most of my trip.  Partners in crime are welcome!

This is my first real hackathon _evar_ although I've spent the last three years social hacking in the Duluth, MN area in an effort to get [Google to bring their FTTH project there](http://www.youtube.com/watch?v=G2i_piWVXuc), and also to attract _large_ data center projects to the region.

I can't seem to decide what I would like to build; there's a good chance I'll throw in with some of you to help you realize and promote your projects.

Looking forward,

Ben Damman

![image](https://f.cloud.github.com/assets/8130/564826/22a90666-c5aa-11e2-80b1-25fd2960fdf7.png)

**Comment by yoni** [comment]
I've created a new API for extracting entities, locations, sentiment and more from petition data. I hope that other folks can use this API to augment their apps with rich data from the petition text.

This is currently in rough form, so I'm eager to get people's feedback, ideas, pull requests and usage examples. I'll continue working on it this weekend, so please expect the API to change A LOT.

The live app and an example call to the API:

http://wetheentities.herokuapp.com/
http://wetheentities.herokuapp.com/petitions/51942b9b2f2c888e2f00000c.json

The codebase and issues:

https://github.com/yoni/wetheentities
https://github.com/yoni/wetheentities/issues

I also opened issue #9 for any discussion and details.

Best,
@yoni

**Comment by theneonlobster** [comment]
Hello. My name is David Platek and I am on the team that built the API. The day of the Hackathon I will be available to answer questions and take comments, and I will also be working on the API itself.

**Comment by sinak** [comment]
Hey folks, I'm Sina (twitter: [sinak](http://twitter.com/sinak)). I started the We the People petition on [unlocking cell phones](https://petitions.whitehouse.gov/petition/make-unlocking-cell-phones-legal/1g9KhZG7) and am now advocating for a fix at [FixtheDMCA](https://petitions.whitehouse.gov/petition/make-unlocking-cell-phones-legal/1g9KhZG7).

I'm best at frontend code and design. Happy to help out with anyone's projects who might need it; feel free to ping me on twitter or email (sina dot khanifar at gmail).

**Comment by sinak** [comment]
Okay, here's a project suggestion: 

**Idea**: A site that allows senator/representatives to see how many of their constituents signed a particular petition.

**Why**: Often petitions on the WtP site require Congressional action. But convincing legislators to take action is difficult, and they generally only care about the opinions of their own constituents, and not the general public. Showing legislators how many of their constituents signed up to a particular petition can have a pretty significant impact on swaying them to taking action.

**Building it**: This could either be done with a server backend, or just via client-side js (similar to the [Petition Mapper](http://kanarinka.brownbag.me/petitionmap/) project). Combining the WtP API with the [Sunlight Congress API](sunlightlabs.github.io/congress/) should be very doable. 

Would love some help in making it happen - feel free to [tweet](http://twitter.com/sinak) or [email me](mailto:sina.khanifar@gmail.com).

This is rather self-serving, as I'm spending the days after the hackathon meeting with staffers in DC to try and convince them that Congress needs to pass a bill to legalize cell phone unlocking. I'd love to have reports that I can print out for each meeting and hand out to them, and I imagine it'd be really useful for other advocates following up on WtP petitions too.

**Comment by jmandzik** [comment]
Hey Sina!

Lot's of overlap between your suggestion and what I'm currently working on.  Would love to join forces and ~~take over the world~~ collaborate. I'm currently working on a few different ways to crunch the zip codes down into sizeable chunks and then query census data.  Hit me up!

**Comment by dlapiduz** [comment]
@sinak @jmandzik I am working on something similar too, also using the Sunlight API. We should chat and maybe combine our efforts!

**Comment by jessebeach** [comment]
@sinak, maybe we can turn #3 into a Dashboard for Congresspeople. Having a specific audience will help us define the feature set more narrowly.

**Comment by sinak** [comment]
@jessebeach Really like that idea. @dlapiduz would love to chat, want to ping me on gchat? i'm sina.khanifar@gmail.com. Combining forces is definitely a win :).

**Comment by entaq** [comment]
Hi all - I am [Arun Nagarajan](http://google.com/+ArunNagarajan) from Google. I work in the NYC office on Google Apps Script. My goal is to build generalized data analysis tools to help journalists and citizens tell better stories from raw data. The product I work on is http://developers.google.com/apps-script. Ideas welcome :)

**Comment by yahelc** [comment]
Hi all, I'm Yahel Carmon, and I'm an analytics developer at <a href="http://bluestatedigital.com">Blue State Digital</a>. @douglasback and I are working on a petition dashboard/analytics project. If you're interested in contributing, come find us. (There seem to be a couple of different approaches bubbling through, though, and that's exciting!)

**Comment by waldoj** [comment]
Folks, at the suggestion of some folks in charge here, I've set up [an organizational GitHub repository for us all to use](https://github.com/organizations/we-the-hackathon/), should anybody care to.

**Comment by ryanatwork** [comment]
Hi I'm [Ryan](http://www.ryanresella.com)  I'm a Senior Engineer at Upworthy.  Former Obama for America engineer, former Technical Lead at Code for America.

**Comment by jessebeach** [comment]
Upload your photos! 

https://plus.google.com/u/0/events/cqm96ov762215hto2n0f67fim9o

**Comment by ivanstegic** [comment]
Such a wonderful time, thanks everyone!

@waldoj thanks for setting up the organization -- are you suggesting we move all repos from yesterday to be under the org? I think it'd be a good way to aggregate them all...

**Comment by yahelc** [comment]
@waldoj if the answer to @ivanstegic's question is yes, I think organization participants need elevated permissions -- it looks like "Participant" isn't high enough to permit forking into the organization.

**Comment by waldoj** [comment]
@ivanstegic Sure, that seemed like a perfectly reasonable possibility. Seems like it'd be good to put them all in one place, or at least copies of them. Thank you, @yahelc—I thought that the permissions were high enough, but apparently administrative privileges are required, which I've just provided. Hopefully nobody now accidentally deletes the organization. :)
